pkg/federate/kubefed/cluster.go

```diff
@@ -14,26 +14,48 @@ import (
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/cache"
 	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/workqueue"
 	"k8s.io/klog"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
 	fedv1 "sigs.k8s.io/kubefed/pkg/apis/core/v1beta1"
 	"sigs.k8s.io/kubefed/pkg/client/generic/scheme"
 )
 
-const (
-	// The size of the event queue buffer for each ClusterEventHandler. This is largely a heuristic - we want a
-	// value reasonably large enough to avoid blocking the sender on bursts (eg when notifying of initial clusters
-	// on ClusterEventHandler registration) but not too large to risk running out of memory due to a slow handler.
-	defaultEventQueueBufferSize = 5000
-	defaultEnqueueTimerInterval = time.Second * 10
-)
+type runnable interface {
+	run()
+}
+
+type clusterEvent struct {
+	handler    federate.ClusterEventHandler
```

Comment:
Why do you need to keep this as state?

You can instead just pass it as a parameter in the run method since you're running it on the clusterWatcher itself so you already have the event handler at hand.

Reply:
sure - either way